Database/Entities/Creature.cs

```diff
@@ -1,24 +1,82 @@
 ﻿namespace Database.Entities;
 
+/// <summary>
+/// Represents an in-game creature and it's data.
+/// Creatures belong to a specific faction and are categorized by tiers, with Tier 1 being the weakest 
+/// and Tier 7 being the strongest. <para/>
+/// 
+/// The <c>Creature</c> class provides properties to store and update the unique identifier, faction id,
+/// tier, base cost, upgrade cost, growth and image data for it's base, upgraded and alternate forms. <para/>
+/// 
+/// The binary image data stored in <c>ImageBytesBase</c>, <c>ImageBytesUpg</c> and <c>ImageBytesUpgAlt</c> 
+/// can be converted to a visual representation using the <c>System.Drawing.Image</c> class. <br/>
+/// Note that <c>System.Drawing</c> might require additional dependencies on some platforms. <para/>
+/// 
+/// <b>Additionally</b>, it includes a <c>Faction</c> entity to which it belongs to.
+/// </summary>
```

Comment:
I love the enthusiasm but this is too much text. The summary is supposed to be a "summary", this is a full-on complete description. Deeper explanations should only be used when it is strictly necessary.

Points of order:

- Anyone working on a project should have some contextual information about it. You can't explain the entire project logic and relations in every single summary.
- The summary should not contain/explain business logic. I.e. whether the creatures belong to a faction or which tier is the strongest is completely irrelevant to the `Creature` class. It is simply a vessel of data. Who uses that data and for what purpose means nothing to the `Creature` class, which this summary is for.
- The fact that this class provides properties to store and update data is kind of the whole idea/premise behind a class. It is akin to saying to a potential car buyer that "This car has four wheels and is used to move you from one place to another". The summary should mainly provide information that is not apparent at first glance.
- Similarly to the previous points, what the image data can be converted to is irrelevant to what the class is or is used for. Not only that but it can also be converted to a thousand different representations, which are not listed here, which (if this was unchanged) would pose the question "why list one but ignore the others?".
- Telling developers that a certain namespace/library "might require additional dependencies on some platforms" is akin to telling a driver "a car might require having its engine started before departing". It is not something that the class should bother with and it is such a fundamental knowledge outlining it is a bad sign (if this wasn't a code review).
- The last paragraph is redundant because of previously mentioned points.

Feel free to look at some examples of summaries, like:

- Built-in .Net classes. On rare occasions some really are a paragraph of text but in those cases there is no way around it.
- The legacy projects (no creature classes there but other summaries which could be useful)
- My `Useful.String.Extensions` github project

Personally, I'd put the following summary:

> Represents a type of unit from the game.